Add inference code

0x434D opened the pull request.

0x434D committed:
Inital commit

Relevant files, script and instructions to perform inference with the presented TIR ControlNet

0x434D committed:
Update README.md

0x434D committed:
Update README.md

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
+# TIR ControlNet
+Christian Mayr, Christian Kübler, Norbert Haala, Michael Teutsch,**"Diffusion based Thermal Semantic Segmentation"**, Perception Beyond the Visual Spectrum Workshop **PBVS**, Computer Vision and Pattern Recognition Conference **CVPR** - 2024, **Poster**
+
+
```
Please add the following line:
"Here is the direct link to the paper: [click me](https://openaccess.thecvf.com/content/CVPR2024W/PBVS/html/Mayr_Narrowing_the_Synthetic-to-Real_Gap_for_Thermal_Infrared_Semantic_Image_Segmentation_CVPRW_2024_paper.html)"

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
-# TIRControlNet
\ No newline at end of file
+# TIR ControlNet
+Christian Mayr, Christian Kübler, Norbert Haala, Michael Teutsch,**"Diffusion based Thermal Semantic Segmentation"**, Perception Beyond the Visual Spectrum Workshop **PBVS**, Computer Vision and Pattern Recognition Conference **CVPR** - 2024, **Poster**
```
The title is
"Narrowing the Synthetic-to-Real Gap for Thermal Infrared Semantic Image Segmentation Using Diffusion-based Conditional Image Synthesis"
and not 
"Diffusion based Thermal Semantic Segmentation"

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
+
+### Image Synthesis Preview
+
+The in the paper presented described approach for IR image synthesis based on provided semantics maps is demonstrated here.
```
The proposed approach for Thermal Infrared (TIR) image synthesis as described in the paper based on provided semantics maps is demonstrated here.

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
+
+## Set Up on Your Own Machine
+
+To allow the research community to further experiment with ControlNet's ability to synthesize high quality IR data, we have decided to release our pretrained weights and inference script. This allows you to synthesize your own IR images.
```
..ControlNet's ability synthesizing high-quality TIR imagery, we provide our pretrained...

MichaelTeutsch replied:
Please replace all occurrences of "IR" with "TIR" to be consistent with the paper.

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
+To allow the research community to further experiment with ControlNet's ability to synthesize high quality IR data, we have decided to release our pretrained weights and inference script. This allows you to synthesize your own IR images.
+
+#### General
+A cuda capable GPU is required to run our code. There you should make sure to have to have your GPU setup correctly.
```
CUDA (all capital letters)

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
+
+#### Virtual Environment
+
+We highly recommend for you to use Conda as a package manager.
```
Remove "for you"

MichaelTeutsch commented on README.md:
````diff
@@ -1 +1,97 @@
+conda install -c conda-forge diffusers transformers accelerate
+```
+
+#### Pretrained weights
````
Pretrained Weights (capital W)

MichaelTeutsch commented on README.md:
````diff
@@ -1 +1,97 @@
+https://huggingface.co/0x434D/TIR_ControlNet
+```
+
+#### Demo images
````
Demo Images (capital I)

MichaelTeutsch commented on README.md:
````diff
@@ -1 +1,97 @@
+```
+
+#### Demo images
+
````
The demo images seem to be missing.

0x434D replied:
Demo images are located under the demo subfolder and work out of the box with the python script - should I mention this here ?

MichaelTeutsch replied:
Probably yes. As a reader I would expect some text or images here after the heading "Demo Images".

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
+The pretrained network was trained on the grayscale semantic maps. That is, the images that directly encode the label IDs. It is critical to use this kind of semantic maps and **not** colored ones.
+
+### Label ID convention - [FMB](https://arxiv.org/abs/2308.02097) Dataset
+The provided pretrained ControlNet has been trained using the [FMB](https://arxiv.org/abs/2308.02097) dataset and and is therefore trained on the conventions contained therein for the definition of label ids. If semantic maps originating from other datasets are supposed to be processed with the provided pretrained model the IDs have to be matched to the ones in the [FMB](https://arxiv.org/abs/230 ...
```
...pretrained TIR ControlNet was trained...

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
+| Bicycle       | 13 | (222, 215, 158)|
+| Pole          | 14 | (135, 113,  90)|
+
+### Size
```
Image Size

MichaelTeutsch commented on README.md:
```diff
@@ -1 +1,97 @@
+
+### Size
+
+Note that the pretrained network is trained using 512 by 512 images - it is advised to stick to this size to achieve the best results.
```
...using images with 512 x 512 pixels each - ...

MichaelTeutsch commented on README.md:
````diff
@@ -1 +1,97 @@
+If this work has been helpful to you, please feel free to cite our paper!
+
+```shell
+
````
@InProceedings{Mayr_2024_CVPR,
    author    = {Mayr, Christian and Kuebler, Christian and Haala, Norbert and Teutsch, Michael},
    title     = {{Narrowing the Synthetic-to-Real Gap for Thermal Infrared Semantic Image Segmentation Using Diffusion-based Conditional Image Synthesis}},
    booktitle = {Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition (CVPR) Workshops},
    month     = {June},
    year      = {2024},
    pages     = {3131--3141}
}

MichaelTeutsch commented on inference.py:
```diff
@@ -0,0 +1,156 @@
+"""
```
Please add license text at the top of the file. See here:
https://github.com/HensoldtOptronicsCV/FineGrainedVesselRecognition/blob/main/download.py

MichaelTeutsch commented on inference.py:
```diff
@@ -0,0 +1,156 @@
+
+
+def pipeline_setup(base_model_path, contronlet_path):
+    """Controlnet pipeline setup
```
ControlNet (capital N)

MichaelTeutsch commented on inference.py:
```diff
@@ -0,0 +1,156 @@
+    return arguments
+
+
+def pipeline_setup(base_model_path, contronlet_path):
```
controlnet_path

MichaelTeutsch commented on inference.py:
```diff
@@ -0,0 +1,156 @@
+            pipe - ControlNet pipeline object for image synthesis
+    """
+    controlnet = ControlNetModel.from_pretrained(
+        contronlet_path,
```
controlnet_path

MichaelTeutsch commented on inference.py:
```diff
@@ -0,0 +1,156 @@
+def parse_arguments():
+    """Define argument parser"""
+    parser = argparse.ArgumentParser(description="Inference script")
+    # Adding positional argument
```
Please better document arguments. Some of them are not fully self-explaining.

Commit:
Add License

0x434D committed:
Updated README.md using PR comments

0x434D committed:
Update inference.py - incorporate PR comments

- Improved argument documentation.
- Renamed all ir to tir for consistency
- Fixed Typos

MichaelTeutsch commented on inference.py:
```diff
@@ -1,7 +1,31 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+# Inspired by https://github.com/avaapm/marveldataset2016/blob/master/MARVEL_Download.py
```
Please remove this line. This was only for the code of Heiko's paper.

MichaelTeutsch commented on inference.py:
```diff
@@ -19,66 +43,75 @@
         help="Prompt for Stable Diffusion",
     )
+
+    # The amount of inference steps ControlNet's synthesis backbone performs to synthesize an image
```
The number of inference steps...

MichaelTeutsch commented on inference.py:
```diff
@@ -19,66 +43,75 @@
         help="Default output folder - if set to None images won\'t be saved",
     )
+
+    # Flag wether to use a random or a set seed for image synthesis
```
whether

0x434D committed:
Update README.md

- Added demo images hint
- Improved Usage description

0x434D committed:
Update inference.py

- Fixed Typos
- Deleted unwanted license line
